eclipsePlugin/src/de/tobject/findbugs/reporter/ThrottledProgressMonitor.java

```diff
@@ -0,0 +1,92 @@
+package de.tobject.findbugs.reporter;
+
+import static org.eclipse.core.runtime.Assert.isNotNull;
+
+import java.util.function.LongSupplier;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+/**
+ * <p>SpotBugs Eclipse Plugin uses {@link IProgressMonitor#setTaskName(String)} to tell what it is working for.
+ * It also updates progress quite frequently, which costs much time to update progress bar.</p>
+ *
+ * <p>Both of them make performance slow, so this class introduces throttling for better performance.</p>
+ *
+ * @author Kengo TODA
+ * @version 3.1
+ */
+@NotThreadSafe
+class ThrottledProgressMonitor implements IProgressMonitor {
+    /**
+     * Span of throttling, in milliseconds.
+     */
+    static final int THROTTLE_SPAN = 300;
+    private static final long NOT_TRIGGERED = Integer.MIN_VALUE;
+
+    private final IProgressMonitor delegate;
+    private final LongSupplier currentTimeMillis;
+    private long lastRenamed = NOT_TRIGGERED;
+    private long lastWorked = NOT_TRIGGERED;
+    private int accumulatedWork;
+
+    ThrottledProgressMonitor(@Nonnull IProgressMonitor delegate, @Nonnull LongSupplier currentTimeMillis) {
+        isNotNull(delegate);
```

Comment:
I personally prefer to read the code like "assert that this is not null" but due the static import it reads now "is not null, so *what*?"
I would prefer to write this like Assert.isNotNull(delegate), which leaves no question what is "isNotNull(delegate)" does.